Sources/TensorFlow/Layer.swift

```diff
@@ -191,3 +191,34 @@ public final class Parameter<Scalar: TensorFlowScalar> {
         self.value = value
     }
 }
+
+/// A wrapper around a differentiable value with "freezable" derivatives.
+///
+/// When `isFrozen` is true, accesses to `value` have a derivative of zero.
+@_propertyWrapper
+public struct Freezable<Value: Differentiable> : Differentiable {
+    @noDerivative public var isFrozen: Bool = false
+    private var _value: Value
```

Comment:
The fact that this is using another level of property delegation makes me think there can be another property wrapper that splits up this logic.

Reply:
I believe the delegation here is necessary only because stored properties can't have custom jvp/vjps. Does it make sense to create a property wrapper to handle that behavior?

Reply:
Definitely! A separate property wrapper should provide a general way to define a custom VJP for a property. I'll write that out in a sec.

Reply:
The current property wrapper implementation in our branch is incomplete—it does not support the both-initialization-and-default-value pattern. I posted a question here: https://forums.swift.org/t/se-0258-property-wrappers-second-review/25843/76?u=rxwei.